crate/operator/scale.py

```diff
@@ -663,16 +663,42 @@ async def handle(  # type: ignore
             apps = AppsV1Api(api_client)
             core = CoreV1Api(api_client)
 
-            await scale_cluster(
-                apps,
-                core,
-                namespace,
-                name,
-                old,
-                scale_master_diff_item,
-                (kopf.Diff(scale_data_diff_items) if scale_data_diff_items else None),
-                logger,
-            )
+            # If old value is zero, resume the cluster (it was suspended)
+            # If new value is zero, suspend the cluster
+            if (
+                # Ensure the diff data exists to keep mypy happy
+                scale_data_diff_items
+                and len(scale_data_diff_items[0]) >= 4
+                # Check for suspend or resume
+                and (
+                    scale_data_diff_items[0][2] == 0 or scale_data_diff_items[0][3] == 0
+                )
+            ):
+                await suspend_or_start_cluster(
```

Comment:
This means that suspending a cluster sends the `IN_PROGRESS` status notification with the text `Suspending cluster and waiting for Persistent Volume Claim(s) to be resized.` to the API [here](https://github.com/crate/crate-operator/blob/master/crate/operator/operations.py#L526), right? Probably we should update this text.

For scaling up it is `Starting cluster. Scaling back up to {new_replicas} nodes. Waiting for node(s) to be present.` which should be fine I think.